SetupDataPkg/Docs/PlatformIntegration/PlatformIntegrationSteps.md

```diff
@@ -83,12 +83,35 @@ the following routines to properly consume configuration data:
 Per silicon policy defintion, platforms are responsible for initializing the silicon policy with a default value when
 configuration when under the circumstance that its corresponding configuration is not present or not even defined.
 
-### Config Data Translation
+### YAML Config Data Translation
 
-If the corresponding configuration is defined and exposed, the platform developer should query variable with the format
+If the corresponding YAML configuration is defined and exposed, the platform developer should query variable with the format
 of `Device.ConfigData.TagID_%08X` or defined as `SINGLE_SETTING_PROVIDER_TEMPLATE`, where the %08X should be populated
 with the intended Tag ID defined in the configuration YAML file set.
 
+### XML Platform Config Knob Shim Library
+
+If following the XML defined configuration, the platform must author a platform level ConfigKnobShimLib that implements
+the generic get knob function from the autogenerated headers from KnobService.py. An example is provided [here](../../Library/Examples/ConfigKnobShimExampleLib/).
+This example library is not intended to be used directly, but as a reference for the platform to implement.
+
+The platform must define `CONF_AUTOGEN_INCLUDE_PATH` in PlatformBuild.py. This is the path that the autogenerated
+header files will be placed under as such: `CONF_AUTOGEN_INCLUDE_PATH\Generated\Config*.h`. The UpdateConfigHdr.py
+build plugin will create the `Generated` directory if it does not exist.
+
+The platform must define `MU_SCHEMA_DIR` and `MU_SCHEMA_FILE_NAME` in PlatformBuild.py. These are the directory
+containing the XML configuration file and the file name of the XML configuration file, respectively. These are split
+apart to allow the CI build to discover a test schema to validate this process.
+
+A platform configuration consumer then includes `<Generated/ConfigClientGenerated.h>` and uses the autogenerated
+specific getter functions to access config knob data.
+
+The platform implemented ConfigKnobShimLib must implement `get_knob_value`, the prototype of which is in
+`ConfigServiceGenerated.h`. ConfigKnobShimLib has the decision whether to return the autogenerated profile defaults
+or to fetch overrides to these from other locations, such as variable storage. Furthermore, the platform
+ConfigKnobShimLib can choose whether to implement config services as a library, linked to each module or as a callout
+via PPI/Protocol to one Config Manager that maintains all platform data in one location.
+
 ## Configuration App Code Integration
 
 1. Ensure all submodules for the platform are based on the latest Project Mu version (e.g. "202108")

```

Comment:
Since we are updating the document, maybe it should suggest the latest release as well. 
(e.g. "202108")
```suggestion
1. Ensure all submodules for the platform are based on the latest Project Mu version (e.g. "release/202208")

```